reactive/coredns.py

```diff
@@ -45,5 +88,40 @@ def start_charm():
 
 @when('layer.docker-resource.coredns-image.changed')
 def update_image():
-    # handle a new image resource becoming available
     clear_flag('charm.coredns.started')
+
+
+@when('charm.coredns.started', 'cluster-dns.domain-ready')
+def send_ip():
+    """
+    Send CoreDNS IP to kuberentes-worker
+    """
+    try:
+        cluster_dns = endpoint_from_flag('cluster-dns.domain-ready')
+        if cluster_dns:
+            service_ip = get_service_ip('coredns')
+            if service_ip:
+                cluster_dns.send_ip(service_ip)
+                clear_flag('cluster-dns.domain-ready')
```

Comment:
This will break the `update_image` handler right? That clears the `charm.coredns.started` flag, but since `cluster-dns.domain-ready` has been cleared, `start_charm` won't actually run again like it's supposed to.

Reply:
True, I was doing this to notify the interface that the flag has changed and that it should change to ready if the Ip was available in [this code](https://github.com/charmed-kubernetes/interface-cluster-dns/blob/master/provides.py#L29).